src/core/namespace.c

```diff
@@ -598,11 +598,24 @@ static int clone_device_node(const char *d, const char *temporary_mount) {
 
         dn = strjoina(temporary_mount, d);
 
-        mac_selinux_create_file_prepare(d, st.st_mode);
-        r = mknod(dn, st.st_mode, st.st_rdev);
-        mac_selinux_create_file_clear();
+        if (do_mknod) {
+                mac_selinux_create_file_prepare(d, st.st_mode);
+                r = mknod(dn, st.st_mode, st.st_rdev);
+                mac_selinux_create_file_clear();
+        } else {
+                _cleanup_close_ int fd = -1;
+
+                mac_selinux_create_file_prepare(d, st.st_mode);
+                fd = open(dn, O_CREAT | O_EXCL | O_CLOEXEC, st.st_mode);
```

Comment:
we can actually use mknod() for creating empty file nodes too, which is what we need here. If we do we can do that in one syscall, without allocating any fd. Which is quicker, simpler and shorter.

Just use `mknod(dn, st.st_mode);`

Reply:
addressed